pyproject.toml

```diff
@@ -63,11 +64,99 @@ include_trailing_comma = true
 force_grid_wrap = 0
 use_parentheses = true
 skip_glob = [
-  "**/connector_builder/generated/**",
-  # TODO: Remove this after we move to Ruff. Ruff is mono-repo-aware and
-  # correctly handles first-party imports in subdirectories.
+    "**/connector_builder/generated/**",
+    # TODO: Remove this after we move to Ruff. Ruff is mono-repo-aware and
+    # correctly handles first-party imports in subdirectories.
 ]
 
+[tool.ruff]
+target-version = "py38"
+line-length = 140
+exclude = [
+    # If any files should not be checked, add them here:
+    "setup.py", # These should be deprecated and will eventually replaced with pyproject.toml-based configuration.
+]
+ignore = [
+    # If any rules should be ignored, add them here:
+    "ANN101", # Missing type annotation for `self` in method
+    "ANN102", # Missing type annotation for `cls` in class method
+    "PT004",  # Add leading underscore to fixtures that do not return anything
+]
+select = [
+    "A",    # flake8-builtins
+    "ANN",  # flake8-annotations
+    "ARG",  # flake8-unused-arguments
+    "B",    # flake8-bugbear
+    "BLE",  # flake8-blind-except
+    "C4",   # flake8-comprehensions
+    "C90",  # mccabe
+    "COM",  # flake8-commas
+    "D",    # pydocstyle/flake8-docstrings
+    "DTZ",  # flake8-datetimezs
+    "E",    # pycodestyle (error)
+    "EM",   # flake8-errmsg
+    "ERA",  # eradicate
+    "F",    # Pyflakes
+    "FA",   # flake8-future-annotations
+    "FBT",  # flake8-boolean-trap
+    "G",    # flake8-logging-format
+    "I",    # isort
+    "ICN",  # flake8-import-conventions
+    "INP",  # flake8-no-pep420
+    "ISC",  # flake8-implicit-str-concat
+    "N",    # pep8-naming
+    "PERF", # perflint
+    "PGH",  # pygrep-hooks
+    "PIE",  # flake8-pie
+    "PLC",  # pylint (convention)
+    "PLE",  # pylint (error)
+    "PLR",  # pylint (refactor)
+    "PLW",  # pylint (warning)
+    "PT",   # flake8-pytest-style
+    "PTH",  # flake8-use-pathlib
+    "Q",    # flake8-quotes
+    "RET",  # flake8-return
+    "RSE",  # flake8-raise
+    "RUF",  # ruff
+    "S",    # flake8-bandit
+    "SIM",  # flake8-simplify
+    "T10",  # flake8-debugger
+    "T20",  # flake8-print
+    "TCH",  # flake8-type-checking
+    "TID",  # flake8-tidy-imports
+    "UP",   # pyupgrade
+    "W",    # pycodestyle (warning)
+    "YTT",  # flake8-2020
+    "SLF",  # flake8-self
+]
+
+[tool.ruff.per-file-ignores]
+"__init__.py" = [
+    "F401", # Permit unused imports in `__init__.py` files
+]
+
+[tool.ruff.flake8-annotations]
+allow-star-arg-any = true
+mypy-init-return = true
+suppress-dummy-args = true
+
+[tool.ruff.flake8-pytest-style]
+parametrize-values-type = "tuple"
+
+[tool.ruff.isort]
+known-first-party = [
+    "airbyte",
+    "airbyte_cdk",
+    "airbyte_protocol",
+    "airbyte_api_client",
+    "connector_ops",
```

Comment:
Ironically this is missing a bunch of those airbyte-ci packages, but I think this is alright. This is enough to start.